core/templates/dev/head/pages/email-dashboard-pages/email-dashboard-data.service.ts

```diff
@@ -16,109 +16,137 @@
  * @fileoverview Services for oppia email dashboard page.
  */
 
-angular.module('oppia').factory('EmailDashboardDataService', [
-  '$http', '$q', function($http, $q) {
-    var QUERY_DATA_URL = '/emaildashboarddatahandler';
-    var QUERY_STATUS_CHECK_URL = '/querystatuscheck';
-    // No. of query results to display on a single page.
-    var QUERIES_PER_PAGE = 10;
-    // Store latest cursor value for fetching next query page.
-    var latestCursor = null;
-    // Array containing all fetched queries.
-    var queries = [];
-    // Index of currently-shown page of query results.
-    var currentPageIndex = -1;
-
-    var fetchQueriesPage = function(pageSize, cursor) {
-      return $http.get(QUERY_DATA_URL, {
-        params: {
-          num_queries_to_fetch: pageSize,
-          cursor: cursor
-        }
-      }).then(function(response) {
-        return response.data;
-      });
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
```

Comment:
Why are you using this HttpHeaders and HttpParams? These are not needed right now since the interceptor is in. 

Interceptor is not working because you haven't added it in this module. You need to add it to `email-dashboard-page.module.ts` similar to how I added it here: https://github.com/oppia/oppia/pull/8228/files for `topics-and-skills-dashboard-page.module.ts`

Reply:
Oh! ok thanks @bansalnitish